new-lamassu-admin/src/pages/UserManagement/modals/EnableUserModal.js

```diff
@@ -0,0 +1,123 @@
+import { useMutation } from '@apollo/react-hooks'
+import { makeStyles } from '@material-ui/core/styles'
+import gql from 'graphql-tag'
+import React, { useState } from 'react'
+
+import Modal from 'src/components/Modal'
+import { Button } from 'src/components/buttons'
+import { Info2, P } from 'src/components/typography'
+
+import styles from '../UserManagement.styles'
+
+import Input2FAModal from './Input2FAModal'
+
+const ENABLE_USER = gql`
+  mutation enableUser($confirmationCode: String, $id: ID!) {
+    enableUser(confirmationCode: $confirmationCode, id: $id) {
+      id
+    }
+  }
+`
+
+const DISABLE_USER = gql`
+  mutation disableUser($confirmationCode: String, $id: ID!) {
+    disableUser(confirmationCode: $confirmationCode, id: $id) {
+      id
+    }
+  }
+`
+
+const useStyles = makeStyles(styles)
+
+const EnableUserModal = ({
+  showModal,
+  toggleModal,
+  user,
+  requiresConfirmation
+}) => {
+  const classes = useStyles()
+
+  const [enableUser] = useMutation(ENABLE_USER, {
+    refetchQueries: () => ['users']
+  })
+
+  const [disableUser] = useMutation(DISABLE_USER, {
+    refetchQueries: () => ['users']
+  })
+
+  const [confirmation, setConfirmation] = useState(null)
+
+  const submit = () => {
+    user?.enabled
+      ? disableUser({
+          variables: {
+            confirmationCode: confirmation,
+            id: user.id
+          }
+        })
+      : enableUser({
+          variables: {
+            confirmationCode: confirmation,
+            id: user.id
+          }
+        })
+    handleClose()
+  }
```

Comment:
I would rather have two functions and make the modal call the correct one based on if the user is enabled or not.